File: JavaScript/프로그래머스/0/181853. 뒤에서 5등까지/뒤에서 5등까지.js
const NUM_RESULT = 5;

function solution(numList) {
    let nList = [...numList];
    let answer = [];
    
    for(let i = 0; i < NUM_RESULT; i++) {
        let minIdx = 0;
        
        nList.forEach((n, j) => {
            if (nList[minIdx] > n) minIdx = j;
        })
        
        answer = [...answer, nList[minIdx]];
        nList.splice(minIdx, 1);
    }
    
    return answer;
}

File: Kotlin/프로그래머스/0/181853. 뒤에서 5등까지/뒤에서 5등까지.kt
class Solution {
    val NUM_RESULT = 5
    
    fun solution(num_list: IntArray): IntArray {
        var n_list = num_list.copyOf()
        var answer = IntArray(NUM_RESULT)
        
        for (i in 0 until NUM_RESULT) {
            var min_idx: Int = i
            for (j in i until n_list.size) {
                if (n_list[min_idx] > n_list[j]) {
                    min_idx = j
                }
            }
            if (i != min_idx) {
                n_list [i] = n_list[min_idx].also {
                    n_list[min_idx] = n_list[i]
                }
            }
            answer[i] = n_list[i]
        }
        return answer
    }
}